lib/models/watcher.js

```diff
@@ -55,22 +59,19 @@ module.exports = class Watcher extends CoreObject {
     }
   }
 
-  _setupBroccoliWatcherBuild() {
-    let heimdallNode;
-
-    this.watcher.currentBuild
-      .then(hash => {
-        heimdallNode = hash.graph.__heimdall__;
-        return hash;
-      })
-      .finally(() => {
-        // guard against `build` rejecting
-        if (heimdallNode) {
-          // remove the heimdall subtree for this build so we don't leak.  If
-          // BROCCOLI_VIZ=1 then we have already output the json in `verboseOutput`.
-          heimdallNode.remove();
-        }
-      });
+  async _setupBroccoliWatcherBuild() {
+    try {
+      const hash = await this.watcher.currentBuild;
+      hash.graph.__heimdall__.remove();
+    } catch (e) {
+      if (e !== null && typeof e === 'object' && e.isBuilderError === true) {
+        // it is safe to ignore build errors here, as this code only exists to
+        // prune heimdall data after a build.
```

Comment:
is it intentional to leave an empty block here? How do you feel about the "minimizing WTFs" factor compared to
```js
const isExpctedError = e !== null && typeof e === 'object' && e.isBuilderError === true; 
if (!isExpctedError) throw e;
// explanation of why we're safe here
```

Reply:
Yes, this is intentional. My preference is the clear `if` `else` in this cases, although I can improve the comment further.